### PR TITLE
Revive the overlay system

### PR DESCRIPTION
**Pull request by andycreeth**
In this PR:
- Expose overlay import/export in a new tab of settings
- Move overlay save/load to use async/await for code tidiness
- Allow importing of an overlay into a new scene collection named after the overlay file
- Overlays now support exporting/importing streamlabels
- Overlays now support exporting/importing widgets

### CODE REVIEW

File: app/services/config-persistence/config.ts [hunk 1]
```diff
@@ -236,12 +236,17 @@ export class ConfigPersistenceService extends PersistentStatefulService<IScenesC
 
 
   switchToBlankConfig(configName = DEFAULT_SCENES_COLLECTION_NAME) {
+    this.switchToBlankConfig(configName);
```
Comment (holiber) [inline app/services/config-persistence/config.ts]: Infinity recursion here. Looks like in some cases app will not start

File: app/services/config-persistence/nodes/overlays/video.ts [hunk 2]
```diff
@@ -15,18 +15,26 @@ interface IContext {
...
   schemaVersion = 1;
 
-
-  save(context: IContext) {
+  async save(context: IContext) {
     const filePath = context.sceneItem.getObsInput().settings['local_file'];
+
+    // Don't bother saving media sources that don't have
+    // a file currently loaded
+    if (!filePath) return;
```
Comment (holiber) [inline app/services/config-persistence/nodes/overlays/video.ts]: What if I want to save filters for the source and add the file a little bit later?

File: app/services/config-persistence/config.ts (hunk 1 above)
Comment (andycreeth) [reply app/services/config-persistence/config.ts]: Yikes, meant to call the new method I created.  Good catch!

File: app/services/config-persistence/nodes/overlays/video.ts (hunk 2 above)
Comment (andycreeth) [reply app/services/config-persistence/nodes/overlays/video.ts]: We don't support that.  The overlay system is meant for exporting finished overlays.  Anything that won't import in a good way for an end user, like a media source without a file, will be ignored.  The main people exporting these files will be our designers.

**Pull request merged**